build/Makefile

```diff
@@ -384,6 +384,7 @@ STATIC_PROVIDERLIB_SRCFILES += \
 	$(PROVIDER_DIR)/support/mycnf-paths.cpp \
 	$(PROVIDER_DIR)/support/productdependencies.cpp \
 	$(PROVIDER_DIR)/support/sqlauth.cpp \
+	$(PROVIDER_DIR)/support/mycnf-scanner.cpp \
```

Comment:
Do you actually need this source file in the provider library? Please verify this.

Given that we can't necessarily read the MySQL configuration files when we're not privileged, I wouldn't expect that you'd need this from the provider itself (the provider runs without privileges). Please investigate, thanks.